packages/build/src/plugins_core/blobs_upload/index.ts

```diff
@@ -6,30 +6,37 @@ import semver from 'semver'
 
 import { log, logError } from '../../log/logger.js'
 import { anyBlobsToUpload, getBlobsDir } from '../../utils/blobs.js'
+import { CoreStep, CoreStepCondition, CoreStepFunction } from '../types.js'
 
 import { getKeysToUpload, getFileWithMetadata } from './utils.js'
 
-const coreStep = async function ({
+const coreStep: CoreStepFunction = async function ({
   debug,
   logs,
   deployId,
   buildDir,
   quiet,
-  constants: { PUBLISH_DIR, SITE_ID, NETLIFY_API_TOKEN, API_URL },
+  packagePath,
+  constants: { SITE_ID, NETLIFY_API_TOKEN, NETLIFY_API_HOST },
 }) {
+  // This should never happen due to the condition check
+  if (!deployId || !NETLIFY_API_TOKEN || !NETLIFY_API_HOST) {
+    return {}
+  }
+
   const storeOpts: { siteID: string; deployID: string; token: string; apiURL: string; fetch?: any } = {
     siteID: SITE_ID,
     deployID: deployId,
     token: NETLIFY_API_TOKEN,
-    apiURL: API_URL,
+    apiURL: NETLIFY_API_HOST.startsWith('http') ? NETLIFY_API_HOST : `https://${NETLIFY_API_HOST}`,
   }
   if (semver.lt(nodeVersion, '18.0.0')) {
     const nodeFetch = await import('node-fetch')
     storeOpts.fetch = nodeFetch.default
   }
 
   const blobStore = getDeployStore(storeOpts)
-  const blobsDir = getBlobsDir({ buildDir, publishDir: PUBLISH_DIR })
+  const blobsDir = getBlobsDir(buildDir, packagePath)
```

Comment:
Any reason why this was changed from accepting an options object to individual args?

Reply:
just two args needed and wont need more in the future so that's easier to type instead of having the `{..}:{...}` 

I think it's an antipattern to use object destruction as parameter on every function. Few arguments are easier to read/type if they have distinct parameters